spec/datetimeformat.html

```diff
@@ -440,6 +462,126 @@ <h1>FormatDateTimeToParts ( _dateTimeFormat_, _x_ )</h1>
+        1. Let _rangePattern_ be *undefined*.
+        1. Let _dateFieldsPracticallyEqual_ be *true*.
+        1. Let _patternContainsLargerDateField_ be *false*.
+        1. While _dateFieldsPracticallyEqual_ is *true* and _patternContainsLargerDateField_ is *false*, repeat for each row of <emu-xref href="#table-datetimeformat-rangepatternfields"></emu-xref> in order, except the header row
+          1. Let _fieldName_ be the name given in the Range Pattern Field column of the row.
+          1. If _fieldName_ is equal to [[AmPm]], then
+            1. Let _rp_ be _rangePatterns_.[[AmPm]].
+            1. If _rangePattern_ is not *undefined* and _rp_ is *undefined*, then
+              1. Set _patternContainsLargerDateField_ to *true*.
+            1. Else,
+              1. Let _v1_ be _tm1_.[[Hour]].
+              1. Let _v2_ be _tm2_.[[Hour]].
+              1. If _v1_ is greater than 11 and _v2_ less or equal than 11, or _v1_ is less or equal than 11 and _v2_ is greater than 11, then
+                1. Set _dateFieldsPracticallyEqual_ to *false*.
+              1. Let _rangePattern_ be _rp_.
+          1. If _fieldName_ is equal to [[FractionalSecondDigits]], then
+            1. Let _rp_ be _rangePatterns_.[[FractionalSecondDigits]].
+            1. If _rangePattern_ is not *undefined* and _rp_ is *undefined*, then
+              1. Set _patternContainsLargerDateField_ to *true*.
+            1. Else,
+              1. Let _fractionalSecondDigits_ be _dateTimeFormat_.[[FractionalSecondDigits]].
+              1. If _fractionalSecondDigits_ is *undefined*, then
+                1. Set _fractionalSecondDigits_ to 3.
+              1. Let _v1_ be _tm1_.[[Millisecond]].
+              1. Let _v2_ be _tm2_.[[Millisecond]].
+              1. Let _v1_ be floor(_v1_ × 10<sup>( _fractionalSecondDigits_ - 3 )</sup>).
+              1. Let _v2_ be floor(_v2_ × 10<sup>( _fractionalSecondDigits_ - 3 )</sup>).
+              1. If _v1_ is not equal to _v2_, then
+                1. Set _dateFieldsPracticallyEqual_ to *false*.
+              1. Let _rangePattern_ be _rp_.
+          1. Else,
+            1. Let _rp_ be _rangePatterns_.[[<_fieldName_>]].
+            1. If _rangePattern_ is not *undefined* and _rp_ is *undefined*, then
+              1. Set _patternContainsLargerDateField_ to *true*.
+            1. Else,
+              1. Let _v1_ be _tm1_.[[<_fieldName_>]].
+              1. Let _v2_ be _tm2_.[[<_fieldName_>]].
+              1. If _v1_ is not equal to _v2_, then
+                1. Set _dateFieldsPracticallyEqual_ to *false*.
+              1. Let _rangePattern_ be _rp_.
+        1. If _dateFieldsPracticallyEqual_ is *true*, then
+          1. Let _pattern_ be _dateTimeFormat_.[[Pattern]].
+          1. Let _patternParts_ be PartitionPattern(_pattern_).
+          1. Let _result_ be ? FormatDateTimePattern(_dateTimeFormat_, _patternParts_, _x_, *undefined*).
+          1. For each _r_ in _result_ do
```

Comment:
```suggestion
          1. For each _r_ in _result_, do
```

`expected "For each" with substeps to end with ", do" (found " do")`